src/main/java/programminglife/model/drawing/DrawableNode.java

```diff
@@ -101,91 +117,102 @@ GenomeGraph getGraph() {
+     * @param width The double representing the width of the DrawableSegment
+     * @param height The double representing the height of the DrawableSegment
+     */
+    final void setSize(double width, double height) {
+        this.setWidth(width);
+        this.setHeight(height);
+        this.setDrawDimensionsUpToDate();
+    }
+
+    /**
      * Set the location to draw this.
      * @param x the x location
      * @param y the y location
      */
-    public void setLocation(double x, double y) {
-        this.location = new XYCoordinate(x, y);
+    public final void setLocation(double x, double y) {
+        this.location.set(x, y);
     }
 
     /**
      * Get the width of the node.
      * @return The width of the node.
      */
-    public double getWidth() {
+    public final double getWidth() {
         return dimensions.getX();
     }
 
     /**
      * Get the height of the node.
      * @return The height of the node.
      */
-    public double getHeight() {
+    public final double getHeight() {
+        if (!isDrawDimensionsUpToDate()) {
+            setDrawDimensions();
+        }
         return dimensions.getY();
     }
 
     /**
      * getter for the center of the left border.
      * @return XYCoordinate.
      */
-    public XYCoordinate getLeftBorderCenter() {
+    public final XYCoordinate getLeftBorderCenter() {
+        if (!isDrawDimensionsUpToDate()) {
+            setDrawDimensions();
+        }
         return new XYCoordinate(location.getX(), location.getY() + 0.5 * getHeight());
     }
 
     /**
      * getter for the center.
      * @return XYCoordinate.
      */
-    public XYCoordinate getCenter() {
+    public final XYCoordinate getCenter() {
+        if (!isDrawDimensionsUpToDate()) {
+            setDrawDimensions();
+        }
         return new XYCoordinate(location.getX() + 0.5 * getWidth(), location.getY() + 0.5 * getHeight());
     }
 
     /**
      * getter for the center of the right border.
      * @return XYCoordinate.
      */
-    public XYCoordinate getRightBorderCenter() {
+    final XYCoordinate getRightBorderCenter() {
+        if (!isDrawDimensionsUpToDate()) {
+            setDrawDimensions();
+        }
         return new XYCoordinate(location.getX() + getWidth(), location.getY() + 0.5 * getHeight());
     }
 
-    /**
-     * Returns the {@link Color} of this DrawableNode.
-     * @return the {@link Color} of this DrawableNode.
-     */
-    public abstract Color getStrokeColor();
-
-    /**
-     * Returns the {@link Color} of this DrawableNode.
-     * @return the {@link Color} of this DrawableNode.
-     */
-    public abstract Color getFillColor();
-
     public final int getIdentifier() {
         return id;
     }
 
-    public XYCoordinate getLocation() {
+    public final XYCoordinate getLocation() {
         return location;
     }
 
-    public Collection<Integer> getGenomes() {
+    public final Collection<Integer> getGenomes() {
         return this.genomes;
```

Comment:
It would probably be useful to override this in dummynodes to return the genomes of the edge